File: samples/server/petstore/springboot/src/main/java/org/openapitools/configuration/EnumConverterConfiguration.java
package org.openapitools.configuration;

import org.openapitools.model.EnumClassDto;
import org.openapitools.model.OuterEnumDto;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.convert.converter.Converter;

/**
 * This class provides Spring Converter beans for the enum models in the OpenAPI specification.
 *
 * By default, Spring only converts primitive types to enums using Enum::valueOf, which can prevent
 * correct conversion if the OpenAPI specification is using an `enumPropertyNaming` other than
 * `original` or the specification has an integer enum.
 */
@Configuration(value = "org.openapitools.configuration.enumConverterConfiguration")
public class EnumConverterConfiguration {

    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.enumClassConverter")
    Converter<String, EnumClassDto> enumClassConverter() {
        return new Converter<String, EnumClassDto>() {
            @Override
            public EnumClassDto convert(String source) {
                return EnumClassDto.fromValue(source);
            }
        };
    }
    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.outerEnumConverter")
    Converter<String, OuterEnumDto> outerEnumConverter() {
        return new Converter<String, OuterEnumDto>() {
            @Override
            public OuterEnumDto convert(String source) {
                return OuterEnumDto.fromValue(source);
            }
        };
    }

}

File: samples/server/petstore/kotlin-spring-issue8366-inheritance-parent-ctor-call/src/main/kotlin/org/openapitools/configuration/EnumConverterConfiguration.kt
package org.openapitools.configuration

import org.openapitools.model.DiscriminatingType

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.convert.converter.Converter

/**
 * This class provides Spring Converter beans for the enum models in the OpenAPI specification.
 *
 * By default, Spring only converts primitive types to enums using Enum::valueOf, which can prevent
 * correct conversion if the OpenAPI specification is using an `enumPropertyNaming` other than
 * `original` or the specification has an integer enum.
 */
@Configuration(value = "org.openapitools.configuration.enumConverterConfiguration")
class EnumConverterConfiguration {

    @Bean(name = ["org.openapitools.configuration.EnumConverterConfiguration.discriminatingTypeConverter"])
    fun discriminatingTypeConverter(): Converter<kotlin.String, DiscriminatingType> {
        return object: Converter<kotlin.String, DiscriminatingType> {
            override fun convert(source: kotlin.String): DiscriminatingType = DiscriminatingType.forValue(source)
        }
    }

}
